packages/itwin/infrastructure-iot/src/components/partials/inputs/InputColorPicker.tsx

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import React, { useEffect, useState } from "react";
+
+import {
+  ColorBuilder,
+  ColorInputPanel,
+  ColorPalette,
+  ColorPicker,
+  ColorSwatch,
+  ColorValue,
+  DropdownMenu,
+  IconButton,
+  MenuExtraContent,
+  useTheme,
+} from "@itwin/itwinui-react";
+
+import styles from "./InputColorPicker.module.scss";
+
+export function InputColorPicker(props: {presetColors?: string[], value: string, onChange: (newValue: string) => void}) {
+
+  // This is needed to iTwin React component to respect system-wide light/dark mode
+  useTheme("os");
+
+  // Storage for color object of input
+  const [color, setColor] = useState<ColorValue>(ColorValue.create(props.value));
+  useEffect(() => {
+    setColor(ColorValue.create(props.value));
+  }, [props.value]);
```

Comment:
The component does re-render when the props change, but the state will not update to the new value without the `useEffect` hook - it will remain set to whatever was the value previously. I have tested this and verified that the hook is needed if you want to correctly update the state after a prop change.